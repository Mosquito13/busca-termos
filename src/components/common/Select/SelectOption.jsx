import classNames from 'classnames';

const SelectOption = ({ icon, text, selected, onSelect, useRoundIconBorder }) => {
  const optionClasses = classNames(
    'select__option',
    selected && 'select__option--selected',
    !onSelect && 'select__option--unselectable'
  );

  const iconClasses = classNames(
    'select__option-icon',
    useRoundIconBorder && 'select__option-icon--rounded-border'
  );

  return (
    <div className={optionClasses} onClick={onSelect}>
      {icon && (
        <div className={iconClasses}>{icon}</div>
      )}
      <div className="select__option-text">{text}</div>
    </div>
  );
};

export default SelectOption;
