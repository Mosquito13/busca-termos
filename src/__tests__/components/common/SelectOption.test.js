import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { FiAnchor } from 'react-icons/fi';

import SelectOption from '../../../components/common/Select/SelectOption';

describe(`${SelectOption.name}`, () => {
  describe('tests', () => {
    let wrapper, onSelectSpy;

    beforeAll(() => {
      onSelectSpy = jest.fn();

      wrapper = shallow(
        <SelectOption text="option text" onSelect={onSelectSpy} />
      );
    });

    it('should render', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.select__option')).toHaveLength(1);
      expect(wrapper.text()).toContain('option text');
    });

    it('should fire onSelect', () => {
      wrapper.find('.select__option').simulate('click');
      expect(onSelectSpy).toBeCalledTimes(1);
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render', () => {
      const tree = renderer
        .create(<SelectOption text="option" onSelect={jest.fn()} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with icon', () => {
      const tree = renderer
        .create(
          <SelectOption
            text="option"
            icon={<FiAnchor />}
            onSelect={jest.fn()}
          />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with icon with rounded borders', () => {
      const tree = renderer
        .create(
          <SelectOption
            text="option"
            icon={<FiAnchor />}
            onSelect={jest.fn()}
            useRoundIconBorder
          />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render selected', () => {
      const tree = renderer
        .create(<SelectOption selected text="option" onSelect={jest.fn()} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render unselectable', () => {
      const tree = renderer.create(<SelectOption text="option" />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
