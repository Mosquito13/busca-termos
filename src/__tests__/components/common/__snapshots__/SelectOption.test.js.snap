// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectOption snapshots should render 1`] = `
<div
  className="select__option"
  onClick={[MockFunction]}
>
  <div
    className="select__option-text"
  >
    option
  </div>
</div>
`;

exports[`SelectOption snapshots should render selected 1`] = `
<div
  className="select__option select__option--selected"
  onClick={[MockFunction]}
>
  <div
    className="select__option-text"
  >
    option
  </div>
</div>
`;

exports[`SelectOption snapshots should render unselectable 1`] = `
<div
  className="select__option select__option--unselectable"
>
  <div
    className="select__option-text"
  >
    option
  </div>
</div>
`;

exports[`SelectOption snapshots should render with icon 1`] = `
<div
  className="select__option"
  onClick={[MockFunction]}
>
  <div
    className="select__option-icon"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="5"
        r="3"
      />
      <line
        x1="12"
        x2="12"
        y1="22"
        y2="8"
      />
      <path
        d="M5 12H2a10 10 0 0 0 20 0h-3"
      />
    </svg>
  </div>
  <div
    className="select__option-text"
  >
    option
  </div>
</div>
`;

exports[`SelectOption snapshots should render with icon with rounded borders 1`] = `
<div
  className="select__option"
  onClick={[MockFunction]}
>
  <div
    className="select__option-icon select__option-icon--rounded-border"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="5"
        r="3"
      />
      <line
        x1="12"
        x2="12"
        y1="22"
        y2="8"
      />
      <path
        d="M5 12H2a10 10 0 0 0 20 0h-3"
      />
    </svg>
  </div>
  <div
    className="select__option-text"
  >
    option
  </div>
</div>
`;
